worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/clipboard/DiskOptimizedClipboard.java

```diff
@@ -169,24 +173,58 @@ public DiskOptimizedClipboard(File file, int versionOverride) {
         nbtMap = new HashMap<>();
         try {
             this.file = file;
+            checkFileLength(file);
             this.braf = new RandomAccessFile(file, "rw");
             braf.setLength(file.length());
             this.nbtBytesRemaining = Integer.MAX_VALUE - (int) file.length();
             init();
-            long biomeLength = (long) ((getHeight() >> 2) + 1) * ((getLength() >> 2) + 1) * ((getWidth() >> 2) + 1);
+
+            int biomeLength = ((getHeight() >> 2) + 1) * ((getLength() >> 2) + 1) * ((getWidth() >> 2) + 1);
+            canHaveBiomes = headerSize + biomeLength < Integer.MAX_VALUE;
```

Comment:
This looks suspicious, `canHaveBiomes` is only false if `headerSize + biomeLength == Integer.MAX_VALUE`? I assume you want an overflow check there?